PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py

```diff
@@ -298,17 +299,15 @@ def miniAOD_customizeCommon(process):
         process.heepIDVarValueMaps.elesMiniAOD = cms.InputTag('reducedEgamma','reducedGedGsfElectrons')
         #force HEEP to use miniAOD (otherwise it'll detect the AOD)
         process.heepIDVarValueMaps.dataFormat = cms.int32(2)
-
-        #add the HEEP trk isol to the slimmed electron
-        process.slimmedElectrons.modifierConfig.modifications[0].electron_config.heepV70TrkPtIso = cms.InputTag("heepIDVarValueMaps","eleTrkPtIso")
-
-    #now put them all in the pat electron
-    process.patElectrons.electronIDSources=cms.PSet()
-    for egid in process.egmGsfElectronIDs.physicsObjectIDs:
-        setattr(process.patElectrons.electronIDSources,egid.idDefinition.idName.value(),cms.InputTag('egmGsfElectronIDs:'+egid.idDefinition.idName.value()))  
-       
-    
+  
+        #add the HEEP trk isol to the slimmed electron, add it to the first FromFloatValMap modifier
+        for pset in process.slimmedElectrons.modifierConfig.modifications:
+            if pset.hasParameter("modifierName") and pset.modifierName == cms.string('EGExtraInfoModifierFromFloatValueMaps'):
+                pset.electron_config.heepV70TrkPtIso = cms.InputTag("heepIDVarValueMaps","eleTrkPtIso")
+                break
```

Comment:
egammaObjectModificationsInMiniAOD_cff steels seems to be a more appropriate place.
- miniAOD_tools should not be exposed to too much detail of individual area modifications
- the coupling between egammaObjectModificationsInMiniAOD_cff and this file already implies that the lists in electron_ids here and the imports done in egammaObjectModificationsInMiniAOD_cff  are consistent

Reply:
First I think that boat has sailed.  Secondly it works slightly differently for the MVAs, they will exist regardless of what IDs are run in VID.  The HEEP is the sole producer which is optionally run and therefore it needs to know if its running. Is this perfect? No. Could this be better? Yes. But its not really hurting things and is it really worth our time to fix?

Reply:
Is "heepV70TrkPtIso" parameter name already encoded inside electron_config via calls to VID tools (like setupAllVIDIdsInModule)?
If not, perhaps dropping "V70" looks appropriate.
IIUC, https://github.com/cms-sw/cmssw/blob/master/PhysicsTools/SelectorUtils/python/tools/vid_id_tools.py#L102
you can actually have multiple HEEPs loaded and heepIDVarValueMaps will make whatever is not heepElectronID-HEEPV60.
This should reduce need to maintain correct naming of this variable.